e2e/testcases/ssh_known_hosts_test.go

```diff
@@ -0,0 +1,204 @@
+		testpredicates.RootSyncHasSourceError(status.SourceErrorCode, "No ED25519 host key is known"),
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+}
+
+func TestRepoReconcilerSSHKnownHost(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.SyncSource, ntopts.SkipNonLocalGitProvider,
+		ntopts.NamespaceRepo(backendNamespace, configsync.RepoSyncName), ntopts.WithDelegatedControl,
+		ntopts.RepoSyncPermissions(policy.AppsAdmin(), policy.CoreAdmin()))
+	var err error
+	repoSecret := &corev1.Secret{}
+	repoReconcilerName := core.NsReconcilerName(backendNamespace, configsync.RepoSyncName)
+	repoSyncNN := nomostest.RepoSyncNN(backendNamespace, configsync.RepoSyncName)
+	nt.T.Cleanup(func() {
+		nt.MustMergePatch(repoSecret, secretDataDeletePatch(controllers.KnownHostsKey))
+		nomostest.DeletePodByLabel(nt, "app", reconcilermanager.ManagerName, true)
+	})
+
+	// validate secret and deployment initial stage
+	err = nt.Validate(nomostest.NamespaceAuthSecretName, backendNamespace, repoSecret)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	err = nt.Watcher.WatchObject(kinds.Deployment(),
+		repoReconcilerName, configsync.ControllerNamespace,
+		[]testpredicates.Predicate{
+			testpredicates.DeploymentHasEnvVar(reconcilermanager.GitSync, controllers.GitSyncKnownHosts, "false"),
+		},
+	)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	// validate repo sync error using invalid known host value
+	knownHostValue := "invalid value"
+	nt.MustMergePatch(repoSecret, secretDataPatch(controllers.KnownHostsKey, knownHostValue))
+	err = nt.Watcher.WatchObject(kinds.Deployment(),
+		repoReconcilerName, configsync.ControllerNamespace,
+		[]testpredicates.Predicate{
+			testpredicates.DeploymentHasEnvVar(reconcilermanager.GitSync, controllers.GitSyncKnownHosts, "true"),
+		},
+	)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	err = nt.Watcher.WatchObject(kinds.RepoSyncV1Beta1(), configsync.RepoSyncName, backendNamespace, []testpredicates.Predicate{
+		testpredicates.RepoSyncHasSourceError(status.SourceErrorCode, "No ED25519 host key is known"),
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	// get known host key value
+	knownHostValue, err = nomostest.GetKnownHosts(nt)
+	if err != nil {
+		nt.T.Fatalf("error: %s", err)
+	}
+
+	// apply known host key and validate
+	nt.MustMergePatch(repoSecret, secretDataPatch(controllers.KnownHostsKey, knownHostValue))
+	err = nt.Watcher.WatchObject(kinds.Deployment(),
+		repoReconcilerName, configsync.ControllerNamespace,
+		[]testpredicates.Predicate{
+			testpredicates.DeploymentHasEnvVar(reconcilermanager.GitSync, controllers.GitSyncKnownHosts, "true"),
+		},
+	)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	// try syncing resource and validate
+	deploymentPath := nomostest.StructuredNSPath(backendNamespace, "deployment-helloworld")
+	deployment := loadDeployment(nt, "../testdata/deployment-helloworld.yaml")
```

Comment:
nits:
- use unstructured
- just use a simple inline object, such as a basic ConfigMap. We don't really care what sort of object is synced here, so it's preferable to use something simple/inline. It's harder to read if you need to reference another file to see the object

Reply:
changed both test use and validate config map instead